src/util/Timer.cpp

```diff
@@ -297,6 +294,21 @@ VirtualClock::crank(bool block)
     return nWorkDone;
 }
 
+void
+VirtualClock::post(std::function<void()> f)
+{
+    std::unique_lock<std::mutex> lock(mExecuteQueueMutex);
+
+    if (mExecuteQueue.empty())
```

Comment:
Adding mDelayExecution flag reverses all the benefits that this PR made. I'm not sure why, but now it again can have big bucket-apply series when it does not accept any network input.

I'll see if re-adding batching fixes that issue.

Reply:
Re-adding batching does not fix that. I'll investigate that properly on Monday.

Reply:
the flag without batching should be functionally equivalent to your original version, so something else is happening :)

Reply:
(especially if you kept the post for the "empty callback")

Reply:
with a batch size of some large number (like 1000) events, it should behave pretty much like the non batching version for test cases

Reply:
I don't have your updated version, but it could be that when the `delay` flag is set you should never post directly to `IOService` ; you can only do that when the flag is not set which can happen when posting from a different thread, as for the main thread code runs from within `poll()` already (and when we call `poll` the flag is always set)

Reply:
could it be that there is an edge cas when processing and event with `mIOService.run_one()` ?

Reply:
Maybe. Another runs does not support my previous claims. I'll make another runs with small modifications.